client/components/comment-editor.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <div>
+    <div ref="commentBody" class="comment-editor"/>
+    <link-dialog ref="linkDialog" @insertLink="onLinkInserted($event)" @removeLink="onLinkRemoved()"/>
+  </div>
+</template>
+
+<script>
+
+  import {EditorState} from 'prosemirror-state';
+  import {EditorView} from 'prosemirror-view';
+  import {DOMParser, DOMSerializer} from "prosemirror-model";
+  import {baseKeymap, toggleMark} from "prosemirror-commands";
+  import {undo, redo, history} from 'prosemirror-history';
+  import {keymap} from 'prosemirror-keymap';
+  import {schema} from './utils/comment-schema.js';
+  import {placeholderPlugin} from './utils/placeholder.js';
+  import {_toggleLink, _clearLink} from './utils/link.js';
+
+  // @vue/component
+  const component = {
+    props: {
+      comment: {
+        type: Object,
+        required: true,
+      },
+      readOnly: {
+        type: Boolean,
+        required: true,
+      },
+    },
+
+    data() {
+      return {
+        isEmpty: true,
+      };
+    },
+
+    mounted() {
+      if (this.readOnly) {
+        this.createReadOnlyEditor();
+      }
+      else {
+        this.createEditor();
+      }
+    },
+    methods: {
+      createEditor() {
+        const state = EditorState.create({
+          schema,
+          plugins: [
+            keymap({
+              'Mod-z': undo,
+              'Shift-Mod-z': redo,
+              'Mod-b': toggleMark(schema.marks.strong),
+              'Mod-i': toggleMark(schema.marks.em),
+              'Mod-u': toggleMark(schema.marks.strikethrough),
+              'Mod-k': _toggleLink(this.$refs.linkDialog),
+            }),
+            keymap(baseKeymap),
+            history(),
+            placeholderPlugin(this, "", this.comment.dummy ? this.$gettext("comment-hint") : this.$gettext("comment-reply-hint")),
+          ],
+        });
+        this.$commentEditorView = new EditorView({mount: this.$refs.commentBody}, {
+          state,
+          dispatchTransaction: (transaction) => {
+            const newState = this.$commentEditorView.state.apply(transaction);
+            this.$commentEditorView.updateState(newState);
+            this.$commentEditorView.state = newState;
+
+            const fragment = DOMSerializer.fromSchema(schema).serializeFragment(newState.doc.content);
```

Comment:
What? What? What is this? Why are you generating a HTML snippet, which you then store in the database, and which you then read back, parse it back into ProseMirror, for ProseMirror to render?

I would understand if you would generate HTML snippet (on server side!) and store it to render it directly. Or if you would store directly document state into the database, and then load it on the client to use ProseMirror to render it in read-only mode. But not both.